base/samples/Transformation.hpp

```diff
@@ -0,0 +1,90 @@
+#ifndef BASE_SAMPLES_TRANSFORMATION_HPP
+#define BASE_SAMPLES_TRANSFORMATION_HPP
+
+#include <string>
+#include <base/Time.hpp>
+#include <base/TransformWithCovariance.hpp>
+#include <base/samples/RigidBodyState.hpp>
+
+namespace base { namespace samples {
+
+/**
+ * Represents a transformation in 3D of a local coordinate frame inside a global coordinate frame.
+ * A frame ID shall be globally unique and corresponds to a coordinate system.
+ * Following the [Rock's conventions](http://rock.opendfki.de/wiki/WikiStart/Standards)
+ * coordinate systems are right handed with X forward, Y left and Z up.
+ *
+ * The transformation \f$ ^{global}_{local}T \f$ in this structure transforms a vector \f$ v_{local} \f$
+ * from the local into the global frame:
+ * \f$ v_{global} = ^{global}_{local}T \cdot v_{local} \f$
+ * In other words the transformation is the pose of the local frame expressed in the global frame.
+ * Example of a transformation chain:
+ * \f$ ^{world}_{sensor}T = ^{world}_{body}T \cdot ^{body}_{sensor}T \f$
+ */
+class Transformation
+{
+public:
+    /** Reference timestamp of the transformation sample */
+    base::Time time;
+
+    /** ID of the global reference frame */
+    std::string global_frame_id;
```

Comment:
> Regarding the docu of RBS

Honestly, if the docu of RBS looks weird, I would personally assume that the problem is that it probably does not match the actual use of RBS. All RBSes I've presonally seen do represent the pose of source in target. For velocities, my experience is that there are as many conventions as people actually generating RBSes.